asserts/header_checks.go

```diff
@@ -201,3 +201,14 @@ func checkStringList(headers map[string]interface{}, name string) ([]string, err
 	}
 	return res, nil
 }
+
+func checkAccountKeyName(headers map[string]interface{}, name string) (string, error) {
```

Comment:
Oops, sorry ... it seems my suggestion was a bit ambiguous. I really just meant to extract something like `checkStringMatches(headers map[string]interface{}, name, pattern string)`. afaik, header_checks.go only contains type-agnostic checks but this is sort-of specific to an account-key.

Having said that, I'm not sure if a `checkStringMatches` should require the header or not.